ui/src/logs/containers/LogsPage.tsx

```diff
@@ -269,17 +295,180 @@ class LogsPage extends Component<Props, State> {
+
+    this.interval = window.setInterval(
+      this.handleTailFetchingInterval,
+      DEFAULT_TAIL_CHUNK_DURATION_MS
+    )
+
+    this.setState({liveUpdating: true})
+  }
+
+  private handleTailFetchingInterval = async () => {
+    if (this.isClearing) {
+      return
+    }
+
+    this.props.executeHistogramQueryAsync()
+    await this.fetchTail()
+  }
+
+  private fetchTail = async () => {
+    await this.props.fetchTailAsync()
+  }
+
+  private fetchNewerChunk = async (): Promise<void> => {
+    const maxNewerFetchForward = moment()
+      .add(DEFAULT_NEWER_CHUNK_DURATION_MS, 'milliseconds')
+      .utc()
+      .valueOf()
+
+    if (this.props.nextNewerLowerBound > maxNewerFetchForward) {
+      this.props.setNextNewerLowerBound(Date.now())
+      this.currentNewerChunksGenerator.cancel()
+    }
+
+    await this.props.fetchNewerChunkAsync()
+  }
+
+  private get isClearing(): boolean {
+    switch (this.props.searchStatus) {
+      case SearchStatus.Clearing:
+      case SearchStatus.None:
+        return true
+    }
+
+    return false
+  }
+
+  private totalForwardValues = (): number => {
+    return getDeep<number | null>(
+      this.props,
+      'tableInfiniteData.forward.values.length',
+      null
+    )
+  }
+
+  private totalBackwardValues = (): number => {
+    return getDeep<number | null>(
+      this.props,
+      'tableInfiniteData.backward.values.length',
+      null
+    )
+  }
+
+  private fetchOlderChunk = async () => {
+    await this.props.fetchOlderChunkAsync()
+  }
+
+  private handleFetchOlderChunk = async () => {
+    if (this.currentOlderChunksGenerator) {
+      return
+    }
+    await this.startFetchingOlder()
+  }
+
+  private handleFetchNewerChunk = async () => {
+    if (this.isLiveUpdating || this.shouldLiveUpdate) {
+      return
+    }
     this.loadingNewer = true
-    this.props.fetchNewerLogsAsync(time)
+    await this.startFetchingNewer()
+  }
+
+  private startFetchingNewer = async () => {
+    if (this.currentNewerChunksGenerator) {
+      return
+    }
+
+    const chunkOptions = {
+      ...NEWER_CHUNK_OPTIONS,
+      getCurrentSize: this.totalForwardValues,
+    }
+
+    this.currentNewerChunksGenerator = fetchChunk(
+      this.fetchNewerChunk,
+      chunkOptions
+    )
+
+    try {
+      await this.currentNewerChunksGenerator.promise
+    } catch (error) {
+      console.error(error)
+    }
+
+    if (!this.currentNewerChunksGenerator.isCanceled) {
+      this.currentNewerChunksGenerator.cancel()
+    }
+    this.loadingNewer = false
+    this.currentNewerChunksGenerator = null
+  }
+
+  private startFetchingOlder = async () => {
+    const chunkOptions = {
+      ...OLDER_CHUNK_OPTIONS,
+      getCurrentSize: this.totalBackwardValues,
+    }
+
+    this.currentOlderChunksGenerator = fetchChunk(
+      this.fetchOlderChunk,
+      chunkOptions
+    )
+
+    try {
+      await this.currentOlderChunksGenerator.promise
+    } catch (error) {
+      console.error(error)
```

Comment:
These errors get logged anyway, so I think the try catch block means they end up getting logged twice.

Reply:
good catch. i updated `fetchEachAsync` to throw its error so that this is handled here only once